src/lib/stores/item.svelte.ts

```diff
@@ -1,86 +1,198 @@
+  selectItem: (storage: string, index: number) => void
+  storage: (storageName: string) => StorageOperations
 }
 
-export async function createItemStore(storagePlaceName: string) {
-  const items = await db.foodItems.where('storage').equals(storagePlaceName).toArray()
+interface StorageOperations {
+  getItemById: (id: string) => StoredItem | undefined
+  getItemByName: (name: string) => StoredItem | undefined
+  addItem: (item: AddItem) => Promise<void>
+  deleteItem: (id: string) => Promise<void>
+  updateItem: (item: UpdateItem) => Promise<void>
+  sortItems: (sortBy: SortBy) => void
+}
 
-  const list = $state<StoredItem[]>(items)
-  let selected = $state(-1)
+export const itemStore = createItemStore()
 
-  return {
-    get list() {
-      return list
-    },
-    add(input: string) {
-      if (input === '')
-        return
+function createItemStore(): ItemStore {
+  const items = $state<Record<string, StoredItem[]>>({})
+  const storages = $derived(Object.keys(items))
+  const itemCounts = $derived.by(() => {
+    const counts: Record<string, number> = {}
+    for (const storage of storages)
+      counts[storage] = items[storage]?.length ?? 0
+    return counts
+  })
+  const storageCount = $derived(storages.length)
+  let selected = $state<{ storage: string, index: number }>({ storage: '', index: -1 })
 
-      let name = input
-      let quantity = 1
+  async function loadInitialData() {
+    const storedItems = await db.foodItems.toArray()
 
-      const itemList = input.split(' ')
-      if (itemList.length > 1 && itemList[0].match(/^\d+$/)) {
-        name = input.slice(itemList[0].length).trim()
-        quantity = Math.min(Number(itemList[0]), 99)
-      }
+    storedItems.forEach((item) => {
+      if (!items[item.storage])
+        items[item.storage] = []
 
-      const newItem: StoredItem = {
-        id: uuid(), // Generate UUID here
-        name,
-        quantity,
-        dateAdded: dayjs().format('YYYY-MM-DD'),
-        shelfLife: 5,
-        storage: storagePlaceName,
-      }
+      items[item.storage].push(item)
+    })
+  }
 
-      db.foodItems.add(newItem)
-      list.push(newItem)
-      selected = list.length
-    },
-    importItem(item: StoredItem) {
-      if (item.storage !== storagePlaceName)
-        throw new Error(`Imported item's storage ${item.storage} did not match destination ${storagePlaceName}`)
+  loadInitialData()
 
-      db.foodItems.add(item)
-      list.push(item)
-      selected = list.length
+  const store: ItemStore = {
+    get items() { return items },
+    get storages() { return storages },
+    get itemCounts() { return itemCounts },
+    get storageCount() { return storageCount },
+    get selected() { return selected },
+    async addStorage(storage: string) {
+      if (!storages.includes(storage)) {
+        storages.push(storage)
+        items[storage] = []
+        await db.foodItems.where('storage').equals(storage).toArray()
```

Comment:
is the result of this supposed to be stored anywhere?

Reply:
hm no that can be removed it seems. i'm still not 100% sure how dexie works